src/site/stages/build/plugins/add-id-to-subheadings.js

```diff
@@ -17,13 +13,11 @@ function createUniqueId(headingEl) {
     .replace(/-+$/, '')
     .substring(0, length);
 
-  if (usedHeaders.includes(anchor)) {
-    if (!usedHeaders.includes(`${anchor}-${currentId}`)) {
-      anchor = `${anchor}-${currentId}`;
-      currentId++;
-    }
+  if (headingOptions.previousHeadings.includes(anchor)) {
```

Comment:
This could throw a runtime error if `headingOptions` or `headingEl` are passed with data structures we're not expecting. Could we do a null check for `headingOptions` key-value pairs and `headingEl` with Lodash/optional chaining before this and warn if any necessary ones are missing? :)

Reply:
`createUniqueId` is just a helper that isn't exported from the module and is only called once, so IMO it isn't worth adding the complexity to validate the parameters. If it did run into a runtime error, the pull request changing this plugin would fail and wouldn't allow merge 🙂